azurerm/internal/services/iottimeseriesinsights/resource_iot_time_series_insights_reference_data_set.go

```diff
@@ -0,0 +1,245 @@
+				),
+			},
+
+			"data_string_comparison_behavior": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  string(timeseriesinsights.Ordinal),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(timeseriesinsights.Ordinal),
+					string(timeseriesinsights.OrdinalIgnoreCase),
+				}, false),
+			},
+
+			"key_property": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(timeseriesinsights.ReferenceDataKeyPropertyTypeBool),
+								string(timeseriesinsights.ReferenceDataKeyPropertyTypeDateTime),
+								string(timeseriesinsights.ReferenceDataKeyPropertyTypeDouble),
+								string(timeseriesinsights.ReferenceDataKeyPropertyTypeString),
+							}, false),
+						},
+					},
+				},
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmIoTTimeSeriesInsightsReferenceDataSetCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).IoTTimeSeriesInsights.ReferenceDataSetsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	environmentName := d.Get("environment_name").(string)
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	resourceGroup := d.Get("resource_group_name").(string)
+	t := d.Get("tags").(map[string]interface{})
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, environmentName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for presence of existing IoT Time Series Insights Reference Data Set %q (Resource Group %q): %s", name, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_iot_time_series_insights_reference_data_set", *existing.ID)
+		}
+	}
+
+	dataset := timeseriesinsights.ReferenceDataSetCreateOrUpdateParameters{
+		Location: &location,
+		Tags:     tags.Expand(t),
+		ReferenceDataSetCreationProperties: &timeseriesinsights.ReferenceDataSetCreationProperties{
+			DataStringComparisonBehavior: timeseriesinsights.DataStringComparisonBehavior(d.Get("data_string_comparison_behavior").(string)),
+			KeyProperties:                expandIoTTimeSeriesInsightsReferenceDataSetKeyProperties(d.Get("key_property").(*schema.Set).List()),
+		},
+	}
+
+	_, err := client.CreateOrUpdate(ctx, resourceGroup, environmentName, name, dataset)
+	if err != nil {
```

Comment:
qwe could comebine these
```suggestion
	if _, err := client.CreateOrUpdate(ctx, resourceGroup, environmentName, name, dataset); err != nil {
```